src/query/plan/operator.cpp

```diff
@@ -2003,7 +2018,17 @@ bool Delete::DeleteCursor::Pull(Frame &frame, ExecutionContext &context) {
   for (TypedValue &expression_result : expression_results) {
     if (MustAbort(context)) throw HintedAbortError();
     if (expression_result.type() == TypedValue::Type::Edge) {
-      auto maybe_value = dba.RemoveEdge(&expression_result.ValueEdge());
+      auto &ea = expression_result.ValueEdge();
+      if (context.auth_checker &&
+          !(context.auth_checker->Accept(*context.db_accessor, ea, auth::FineGrainedPermission::CREATE_DELETE) &&
+            context.auth_checker->Accept(*context.db_accessor, ea.To(), storage::View::NEW,
+                                         auth::FineGrainedPermission::CREATE_DELETE) &&
+            context.auth_checker->Accept(*context.db_accessor, ea.From(), storage::View::NEW,
+                                         auth::FineGrainedPermission::CREATE_DELETE))) {
```

Comment:
Does the user need `CREATE_DELETE` permission for a vertex in order to delete one of its edges?

Reply:
If this is done in order to make sure the the user has access to the vertices they want to delete, then I might run the authorization check for the vertices before deleting the edges and use `UPDATE` permission here for vertices.